src/main/java/htsjdk/samtools/filter/ReadNameFilter.java

```diff
@@ -81,17 +81,7 @@ public ReadNameFilter(final Set<String> readNameFilterSet, final boolean include
      */
     @Override
     public boolean filterOut(final SAMRecord record) {
-        if (includeReads) {
-            if (readNameFilterSet.contains(record.getReadName())) {
-                return false;
-            }
-        } else {
-            if (!readNameFilterSet.contains(record.getReadName())) {
-                return false;
-            }
-        }
-
-        return true;
+        return readNameFilterSet.contains(record.getReadName()) != includeReads;

```

Comment:
While this version is more compact, I also think it is more confusing -- We had to draw a truth table to confirm that the behavior is the same in all cases.